LameDLLWrap/LibMp3Lame.cs

```diff
@@ -639,6 +674,16 @@ public bool ID3SetFieldValue(string value)
 			return CheckResult(NativeMethods.id3tag_set_fieldvalue_utf16(context, data));
 		}
 
+		public bool ID3SetTextInfo(string id, string text)
+		{
+			if (Encoding.UTF8.GetByteCount(text) == text.Length)
+				return CheckResult(NativeMethods.id3tag_set_textinfo_latin1(context, id, text));
+
+			// Value is Unicode.  Encode as UCS2 with BOM and terminator.
+			byte[] data = UCS2.GetBytes(text);
+			return CheckResult(NativeMethods.id3tag_set_textinfo_utf16(context, id, data));
+		}
```

Comment:
All of the `ID3SetXXX` methods are checking the UTF8 byte count then calling this... which then checks the byte count. Might as well fix up the Latin1 comparison here (round-trip through Encoding.Latin1 and compare result to `text`) and then change all of the `ID3SetXXX` methods to call this directly instead of checking the string first.